admin_tests/regisration_schemas/test_views.py

```diff
@@ -0,0 +1,198 @@
+
+        context = view.get_context_data()
+        assert context['registration_schema'] == registration_schema
+        assert context['form'].data['active'] == registration_schema.active
+        assert context['form'].data['visible'] == registration_schema.visible
+
+    def test_registration_schema_update(self, view, registration_schema):
+        assert not registration_schema.visible
+        assert not registration_schema.active
+        form = view.get_form()
+
+        # `['on'] indicates a selected toggle from this form
+        form.data['active'] = ['on']
+        form.data['visible'] = ['on']
+
+        view.form_valid(form)
+
+        registration_schema.refresh_from_db()
+        assert registration_schema.visible
+        assert registration_schema.active
+
+
+@pytest.mark.django_db
+@pytest.mark.urls('admin.base.urls')
+class TestCreateRegistrationSchema:
+
+    @pytest.fixture()
+    def req(self):
+        return RequestFactory().get('/fake_path')
+
+    @pytest.fixture
+    def csv_data(self):
+        return b'block_type,display_text,help_text,example_text,required,registration_response_key,NOEX_updates,' \
+               b'NOEX_update_reason\npage-heading,This is the page heading,"This is extra, helpful context",,FALSE,,' \
+               b'FALSE,'
+
+    @pytest.fixture
+    def csv_file(self, csv_data):
+        return SimpleUploadedFile('test_file.csv', csv_data, content_type='application/csv')
+
+    @pytest.fixture()
+    def view(self, req):
+        plain_view = views.RegistrationSchemaCreateView()
+        view = setup_view(plain_view, req)
+        return view
+
+    def test_registration_schema_create(self, view, csv_file, req):
+        # django.contrib.messages has a bug which effects unittests
+        # more info here -> https://code.djangoproject.com/ticket/17971
+        setattr(req, 'session', 'session')
+        messages = FallbackStorage(req)
+        setattr(req, '_messages', messages)
+
+        form = view.get_form()
+
+        form.data['name'] = 'Trust the Process'
+        form.files['schema'] = csv_file
+
+        view.form_valid(form)
+
+        registration_schema = RegistrationSchema.objects.get(name='Trust the Process')
+        assert registration_schema.schema_blocks.count() == 1
+        block = registration_schema.schema_blocks.first()
+        assert block.block_type == 'page-heading'
+        assert block.display_text == 'This is the page heading'
+        assert registration_schema.schema_version == 1
+
+        view.form_valid(form)
+        registration_schema = RegistrationSchema.objects.get_latest_version(name='Trust the Process')
+        assert registration_schema.schema_version == 2
+
+    def test_registration_schema_csv_to_blocks(self, view, csv_file):
+        blocks = view.csv_to_blocks(csv_file)
+        assert len(blocks) == 1
+        assert blocks[0]['block_type'] == 'page-heading'
+        assert blocks[0]['display_text'] == 'This is the page heading'
```

Comment:
nit: Consider three tests

One to test the creation of database entities (RegistrationSchema and underlying blocks just exist in the correct number)
One to test the content when they're created (which you're doing twice now by testing it in the create and additionally testing the underlying `csv_to_blocks` function)
One to test the version incrementing.